File: SpringBoot/src/main/java/com/hospital/qlbv/entity/Patient.java
package com.hospital.qlbv.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;

@Entity
@Table(name = "patients")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Patient {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer patientId;
    
    @Column(name = "user_id")
    private Integer userId;
    
    @Column(name = "created_at")
    private String createdAt;
    
    private String status;
    private String bloodType;
    private String allergies;
    private BigDecimal height;
    private BigDecimal weight;
    
    @Column(name = "insurance_id")
    private Integer insuranceId;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", insertable = false, updatable = false)
    @JsonIgnoreProperties({"passwordHash", "userRoles", "email", "phone", "fullName"})
    private User user;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "insurance_id", insertable = false, updatable = false)
    @JsonIgnoreProperties("patients")
    private Insurance insurance;
}

File: Backend/Models/Patient.cs
using System.ComponentModel.DataAnnotations;

namespace QuanLyBenhVien.API.Models;

public class Patient
{
    [Key]
    public int PatientID { get; set; }
    public int? UserID { get; set; }
    public DateTime? CreatedAt { get; set; }
    public string? Status { get; set; }
    public string? BloodType { get; set; }
    public string? Allergies { get; set; }
    public decimal? Height { get; set; }
    public decimal? Weight { get; set; }
    public int? InsuranceID { get; set; }

    // Navigation properties
    public virtual User? User { get; set; }
    public virtual Insurance? Insurance { get; set; }
    public virtual ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();
    public virtual ICollection<MedicalRecord> MedicalRecords { get; set; } = new List<MedicalRecord>();
    public virtual ICollection<Payment> Payments { get; set; } = new List<Payment>();
}
